lib/src/main/scala/spinal/lib/bus/amba3/apb/Apb3SlaveFactory.scala

```diff
@@ -38,19 +38,25 @@ class Apb3SlaveFactory(bus: Apb3, selId: Int, dontCareReadData : Boolean = false
   bus.PREADY := True
   if(dontCareReadData) bus.PRDATA.assignDontCare() else bus.PRDATA := 0
 
-  if(bus.config.useSlaveError) bus.PSLVERROR := False
+  val writeErrorFlag = False
+  val readErrorFlag = False
 
   val askWrite = (bus.PSEL(selId) && bus.PENABLE && bus.PWRITE).allowPruning()
   val askRead  = (bus.PSEL(selId) && bus.PENABLE && !bus.PWRITE).allowPruning()
   val doWrite  = (bus.PSEL(selId) && bus.PENABLE && bus.PREADY &&  bus.PWRITE).allowPruning()
   val doRead   = (bus.PSEL(selId) && bus.PENABLE && bus.PREADY && !bus.PWRITE).allowPruning()
 
+  if (bus.config.useSlaveError) bus.PSLVERROR := (doWrite && writeErrorFlag) || (doRead && readErrorFlag)
+
   override def readAddress()  = bus.PADDR
   override def writeAddress() = bus.PADDR
 
   override def readHalt()  = bus.PREADY := False
   override def writeHalt() = bus.PREADY := False
 
+  override def readError(): Unit = readErrorFlag := True
+  override def writeError(): Unit = writeErrorFlag := True
```

Comment:
I notice code duplication, would it be possible to add this declaration + logic in extended class or trait?

Reply:
Yes it's duplicated code but it matches the patterns already present in BusSlaveFactory.

Reply:
#933